moveit_core/robot_state/include/moveit/robot_state/attached_body.h

```diff
@@ -110,6 +111,63 @@ class AttachedBody
     return attach_trans_;
   }
 
+  /** \brief Get the fixed transform to a subframe on this body (not a transform to a visual or collision shape).
+   * The frame_name needs to have the object's name prepended (e.g. "screwdriver/tip" returns true if the object's
+   * name is "screwdriver"). Returns an identity transform if frame is not found. 
+   * Use hasSubframeTransform to check if it exists.
+  */
+  const Eigen::Isometry3d& getSubframeTransform(const std::string& frame_name) const
+  {
+    // Check if the frame_name has the object's name prepended
+    int prefix_length = id_.length() + 1;
+    try 
+    {
+      auto i = subframe_poses_.find(frame_name.substr(prefix_length));
+      if (i != subframe_poses_.end())
+        return i->second;
+      throw std::runtime_error("A subframe with name " + frame_name + " was requested from object " + id_ + " but it did not exist.");
+    }
+    catch (std::exception exc) {;}
+  }
+
+  /** \brief Returns true if the subframes contain the frame_name.
+   * The frame_name needs to have the object's name prepended (e.g. "screwdriver/tip" returns true if the object's
+   * name is "screwdriver").
+  */
+  bool hasSubframeTransform(const std::string& frame_name) const
+  {
+    // Check if the frame_name has the object's name prepended
+    int prefix_length = id_.length() + 1;
+    try 
+    {
+      return (subframe_poses_.find(frame_name.substr(prefix_length)) != subframe_poses_.end());
+    }
+    catch (std::exception exc) {;}
+    return false;
+  }
+
+  /** \brief Get all named transforms (not the ones associated to visual or collision shapes)
+   * These are also technically fixed transforms, but changing the original getFixedTransforms function would be bad.
+   */
+  const std::map<std::string, Eigen::Isometry3d>& getSubframeTransforms() const
```

Comment:
Nit (feel free to ignore): IMO `getAllSubframeTransforms()` is better.

Reply:
Wouldn't mind, but `getTransforms()` already existed